### PR TITLE
Updated ChinaUnionPay/Discover Regex

### PR DESCRIPTION
**Pull request by limianwang**
Updated the Discover Regex. The old one did not include China Union Pay, which is part of Discover now. 

The following is taken from: http://www.richardsramblings.com/regex/credit-card-numbers/

> Discover Card
> 
> The first six digits of any credit or debit card identify the issuing authority or bank. According to the Discover Network’s developer documentation, Discover Card’s issuer identification numbers start with 6011, 622126-622925 (Discover cards in this range are dual-branded with UnionPay), 644-649, or 65.
> 
> Not as intuitively simple as VISA or MasterCard regexes due to the full-length IINs, an input-validation regex for 16-digit Discover card numbers needs to account for the range of six-digit issuer identifiers. No spaces or dashes are allowed, e.g. “6011000990139424″.
> ^6(?:011\d\d|5\d{4}|4[4-9]\d{3}|22(?:1(?:2[6-9]|[3-9]\d)|[2-8]\d\d|9(?:[01]\d|2[0-5])))\d{10}$

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
+
+'use strict';
+
 //Regular expressions are from http://activemerchant.rubyforge.org/classes/ActiveMerchant/Billing/CreditCardMethods.html
+
 var identifiers = [
 	{name:'VISA', pattern:/^4\d{12}(\d{3})?$/},
 	{name:'MasterCard', pattern:/^(5[1-5]\d{4}|677189)\d{10}$/},
+  {name:'Discover', pattern:/^6(?:011\d\d|5\d{4}|4[4-9]\d{3}|22(?:1(?:2[6-9]|[3-9]\d)|[2-8]\d\d|9(?:[01]\d|2[0-5])))\d{10}$/},
```
Comment (VRMink) [inline lib/index.js:9]: This overlaps the regex that's already there for Discover, please rewrite the reqex so that it only catches the additional edgecases.

File: test/card-test.js [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
 		identifyCard('349237692562216').should.eql('Amex')
 		identifyCard('375886564589009').should.eql('Amex')
-	})
-
-})
-
-
+	});
 
+  it('Discover', function() {
```
Comment (VRMink) [inline test/card-test.js:25]: Thank you

**Comment by VRMink** [comment]
Thank you for your contributed. It is very much appreciated, I wasn't aware of this edge case for Discover.

Please, will you rewrite the regex from the article you refer to, so that it only catches the additional edge case and doesn't overlap with the Discover regex that is already there? It would make for better logic and also look much nicer :)

**Comment by limianwang** [comment]
@VRMink I can do that. However, I believe from the article, that would be the regex to cover all Discover Card, unless i'm mistaken. In the test, I've covered a sample Test Card the Discover Site.

edit:

Oups, My changes should've replaced the existing Discover Regex. Sorry. I will fix this and make sure test runs.
